### DISCUSSION TITLE
unsetField(), what exactly is data-felte-keep-on-remove ?

### DISCUSSION THREAD

**Discussion by squiba36** [pablo-abc/felte#190] [Q&A] [answered]
I was trying to reproduce the tag (adding/removing from array) example. But the unsetField() was removing all items from tag array after the given index (including the index item).

The desired behaviour is obtained only after tag input has "data-felte-keep-on-remove".  Can anyone tell what exactly this do and provide link to the relevant source code files.

**Comment by pablo-abc** [comment, marked as answer]
The source code that handles that is [here](https://github.com/pablo-abc/felte/blob/6b3362e70e7370681edd9a97b93b8506676bc9ab/packages/core/src/create-form-action.ts#L228). That attribute basically tells Felte to keep a field on its data object even if the DOM element is removed.

The reason why it can be _finicky_ with `unsetField` is that it _can_ conflict with keyed lists. I believe it causes no issues in Svelte if you leave the `each` block without a key?

Another thing is that the code linked runs in a MutationObserver, which removes the field when the DOM element is removed. Since `unsetField` removes the field from Felte's `data` store, and since the easiest approach to render these lists is by iterating over the values themselves (e.g. `{#each data.items as item}`), adding the `data-felte-keep-on-remove` is necessary for now.

There might be a way to prevent this from being necessary but I have not looked into it.

**Reply by squiba36** [reply]
Yes, it works fine if we leave the each block without the key.
